Add information about releasing stable branches (#783)
submariner-io/submariner-website: src/content/development/release-process/_index.en.md

content for the release, this needs to be reviewed. Once the pull request is merged, the release process will continue automatically
and the next step can be initiated shortly after making sure the release jobs on the `releases` and any participating repositories are done.

> Starting with 0.13, the release process changed.
> In order to properly release stable versions, make sure to run the `make release` commands on a branch based off the stable release
> branch you're releasing for (`release-<major>.<minor>`).

Reviewer: ```suggestion
Starting with 0.13, when creating a release for a stable version (`release-<major>.<minor>`), the `make release` commands must be run on a branch based off the associated stable release branch.
```
